packages/wrangler/src/zones.ts

```diff
@@ -13,9 +13,7 @@ export function getHostFromRoute(route: Route): string | undefined {
 	return typeof route === "string"
 		? getHostFromUrl(route)
 		: typeof route === "object"
-		? "zone_name" in route
-			? getHostFromUrl(route.zone_name)
```

Comment:
are we doing this because `zone_name` is handled elsewhere, or is this a breaking change?

Reply:
It's a bugfix for https://github.com/cloudflare/wrangler2/issues/1814, where the correct subdomain is only detected for routes with `zone_id`, not `zone_name`. I brought it up in a meeting a couple weeks ago, and people seemed to be on board. I'm happy to revisit that though—I suppose on the face of it it is a breaking change

Reply:
Wait it doesn't look like a breaking change since pattern is **always** required right?

Reply:
Yeah okay this change is legit 👍

Reply:
Yeah, pattern is always required. This just means that now the domain will be taken from `pattern` rather than `zone_name` always, rather than just when `zone_id` is provided

Reply:
This now has broken routes with `*/*` which is required for Cloudflare for SaaS, so that the worker is invoked by an external custom hostname on the zone ...

Reply:
I just created a ticket for this as all of our Cloudflare for SaaS workers can no longer deploy #2095